--- a/gcc/genautomata.c
+++ b/gcc/genautomata.c
@@ -1178,7 +1178,7 @@
 	}
     }
   obstack_1grow (&irp, '\0');
-  out_str = obstack_base (&irp);
+  out_str = (char *) obstack_base (&irp);
   obstack_finish (&irp);
 
   *pstr = p;
@@ -6882,7 +6882,7 @@
 {
   form_regexp (regexp);
   obstack_1grow (&irp, '\0');
-  return obstack_base (&irp);
+  return (const char *) obstack_base (&irp);
 }
 
 /* The function frees memory allocated for last formed string
@@ -9298,7 +9298,7 @@
   obstack_grow (&irp, STANDARD_OUTPUT_DESCRIPTION_FILE_SUFFIX,
 		strlen (STANDARD_OUTPUT_DESCRIPTION_FILE_SUFFIX) + 1);
   obstack_1grow (&irp, '\0');
-  output_description_file_name = obstack_base (&irp);
+  output_description_file_name = (char *) obstack_base (&irp);
   obstack_finish (&irp);
 }
 --- a/gcc/godump.c
+++ b/gcc/godump.c
@@ -926,7 +926,7 @@
 
   ob = &container->type_obstack;
   obstack_1grow (ob, '\0');
-  fputs (obstack_base (ob), go_dump_file);
+  fputs ((const char *) obstack_base (ob), go_dump_file);
   obstack_free (ob, obstack_base (ob));
 }
